include/dr/mhp/views/segmented.hpp

```diff
@@ -69,6 +71,9 @@ class segmented_view : public rng::view_base {
                                    rng::end(segments_tpl_));
   }
 
+  auto size() const { return rng::size(segments_tpl_); }
```

Comment:
Inheriting from [view_interface](https://en.cppreference.com/w/cpp/ranges/view_interface) instead of `view_base` is preferred as it will define `size()` and `empty()`. One of us can do that in a separate commit.

https://github.com/oneapi-src/distributed-ranges/blob/3beb3f522cb7f28a0c64926ac782f695a80fbed5/include/dr/shp/zip_view.hpp#L108